### DISCUSSION TITLE
Runtime performance VM google cloud and HCP

### DISCUSSION THREAD

**Discussion by Fadwa7** [sylabs/singularity#2741] [Q&A]
Hi

I've been working on creating a container, and while the build process was successful, I've noticed that the runtime performance of the container is significantly slower in HCP compared to running the tool installed directly on a Google Cloud VM.

Is this discrepancy in runtime performance normal? If not, could you please help me identify the problem?

Below is the recipe I used to create the container:
```
Bootstrap: docker
From: ubuntu:latest

%labels
        authors="Fadwa"
        description="Reindeer Image"
%files
	bcalm.yml
%environment
        PATH=/opt/reindeer/:/opt/reindeer/bin/:$PATH

%post
        apt-get update -y && apt-get install -y build-essential bc wget git unzip
	apt-get install -y cmake-qt-gui
	apt-get install -y zlib1g-dev
	apt-get update -y
        rm -rf /var/lib/apt/lists/*
        mkdir -p /opt/etc
        export DEBIAN_FRONTEND=noninteractive
        export LC_ALL=C.UTF-8
        echo 'export LC_ALL=C.UTF-8' >> "$SINGULARITY_ENVIRONMENT"
        echo 'export LANG=C.UTF-8' >> "$SINGULARITY_ENVIRONMENT" 

	mkdir /opt/reindeer && cd /opt/reindeer &&
        git clone --recursive https://github.com/kamimrcht/REINDEER.git . &&
        sh install.sh

%runscript
	/opt/reindeer/Reindeer "$@"
```
Thank you in advance

**Comment by dtrudg** [comment]
Converted from an issue, as this is a usage discussion rather than a specific bug or feature issue.

What is HCP in this context? There is generally a very small performance overhead from a container, if any at all. If you see a large difference it's likely the environments you are running in are quite different.

You would need to give detail about what the HCP environment is, what the Google Cloud VM is, how you are running the container, and what you are seeing in order for anyone to be able to offer assistance.
